### PR TITLE
Enhance security

### PR DESCRIPTION
**Pull request by edifirut**
Replace local API key with COS API key
Secure all oasis routes
Call message store with credentials and tenant data

### CODE REVIEW

**Review by xvasi — APPROVED** [review]

File: packages/server/channels-api/config/env_config.go [hunk 1]
```diff
@@ -2,11 +2,12 @@ package config
...
 type Config struct {
 	Service struct {
-		MessageStore  string `env:"MESSAGE_STORE_URL,required"`
-		ServerAddress string `env:"CHANNELS_API_SERVER_ADDRESS,required"`
-		GRPCPort      int    `env:"CHANNELS_GRPC_PORT,required"`
-		OasisApiUrl   string `env:"OASIS_API_URL,required"`
-		CorsUrl       string `env:"CHANNELS_API_CORS_URL,required"`
+		MessageStoreUrl    string `env:"MESSAGE_STORE_URL,required"`
+		MessageStoreApiKey string `env:"MESSAGE_STORE_API_KEY,required"`
+		ServerAddress      string `env:"CHANNELS_API_SERVER_ADDRESS,required"`
```
Comment (tsearle) [inline packages/server/channels-api/config/env_config.go:7]: update README with new env vars

File: packages/server/oasis-api/routes/main.go [hunk 2]
```diff
@@ -33,19 +28,23 @@ func getRouter(config *c.Config, fh *FeedHub.FeedHub, mh *MessageHub.MessageHub)
...
-	route.Use(apiKeyChecker(config.Service.ApiKey))
+	route.Use(service.ApiKeyCheckerHTTP(commonRepositories.AppKeyRepo, service.OASIS_API))
+	route.Use(service.UserToTenantEnhancer(commonRepositories.UserToTenantRepo))
 
-	df := util.MakeDialFactory(config)
-	addFeedRoutes(route, config, df)
-	addCallCredentialRoutes(route, config)
+	df := util.MakeDialFactory(conf)
+	addFeedRoutes(route, conf, df)
+	addCallCredentialRoutes(route, conf)
```
Comment (tsearle) [inline packages/server/oasis-api/routes/main.go:36]: update call Credential Route to perform tennant check

**Review by tsearle — CHANGES_REQUESTED** [review]
webrtc security needs to be implemented

File: packages/server/channels-api/config/env_config.go (hunk 1 above)
Comment (edifirut) [reply packages/server/channels-api/config/env_config.go:7]: updated

File: packages/server/oasis-api/routes/main.go (hunk 2 above)
Comment (edifirut) [reply packages/server/oasis-api/routes/main.go:36]: already checked for tenant. i did a change to not pass anymore the username as parameter and use the username form header

**Review by tsearle — APPROVED** [review]

**Pull request merged**